flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/stream/sql/MultiJoinTest.java

```diff
@@ -684,4 +695,705 @@ void testMultiSinkOnMultiJoinedView() {
+                        + "  SUM(s.amount) AS total_revenue, "
+                        + "  AVG(s.amount) AS avg_sale_amount, "
+                        + "  MAX(s.amount) AS max_sale_amount "
+                        + "FROM Users u "
+                        + "LEFT JOIN Orders o ON u.user_id = o.user_id "
+                        + "LEFT JOIN Categories c ON u.user_id = c.user_id AND o.product = c.category_id "
+                        + "LEFT JOIN Sales s ON u.user_id = s.user_id "
+                        + "GROUP BY c.category_name "
+                        + "HAVING COUNT(s.sale_id) > 0");
+    }
+
+    @Test
+    void testSameTableMultipleAliases() {
+        util.verifyRelPlan(
+                "SELECT * "
+                        + "FROM Users u "
+                        + "LEFT JOIN Users u1 ON u.user_id = u1.user_id "
+                        + "LEFT JOIN Users u2 ON u1.user_id = u2.user_id "
+                        + "LEFT JOIN Users u3 ON u2.user_id = u3.user_id");
+    }
+
+    @Test
+    @Tag("expected-multijoin-chain")
+    void testWithExpressionInJoinCondition() {
+        util.tableEnv()
+                .executeSql(
+                        "CREATE TABLE Products ("
+                                + "  product_id STRING PRIMARY KEY NOT ENFORCED,"
+                                + "  price DOUBLE,"
+                                + "  discount DOUBLE"
+                                + ") WITH ('connector' = 'values', 'changelog-mode' = 'I,UA,D')");
+
+        util.tableEnv()
+                .executeSql(
+                        "CREATE TABLE Sales ("
+                                + "  sale_id STRING PRIMARY KEY NOT ENFORCED,"
+                                + "  product_key DOUBLE,"
+                                + "  quantity INT"
+                                + ") WITH ('connector' = 'values', 'changelog-mode' = 'I,UA,D')");
+
+        util.tableEnv()
+                .executeSql(
+                        "CREATE TABLE Promotions ("
+                                + "  promo_id STRING PRIMARY KEY NOT ENFORCED,"
+                                + "  product_key DOUBLE,"
+                                + "  promo_text STRING"
+                                + ") WITH ('connector' = 'values', 'changelog-mode' = 'I,UA,D')");
+
+        util.verifyRelPlan(
+                "SELECT "
+                        + "  p.product_id, "
+                        + "  (p.price - p.discount) AS net_price, "
+                        + "  s.quantity, "
+                        + "  pr.promo_text "
+                        + "FROM Products AS p "
+                        + "LEFT JOIN Sales AS s "
+                        + "  ON (p.price - p.discount) = s.product_key "
+                        + "LEFT JOIN Promotions AS pr "
+                        + "  ON (p.price - p.discount) = pr.product_key "
+                        + "WHERE (p.price - p.discount) > 100");
+    }
+
+    @Test
+    @Tag("no-common-join-key")
```

Comment:
Also, what would be the difference between no-common-join-key and multi-join-chain? This one has also multiple multijoins